src/app/template-list/template-list.component.ts

```diff
@@ -64,8 +64,18 @@ export class TemplateListComponent implements OnInit {
       this.router.navigate(['template',1])
     }
 
-    
-    
+    removeTemplate(tmpl): void {
+      this.templatesService.currTmpl = tmpl;
+      this.templatesService
+      .removeTemplate(tmpl).
+      then(() => {
+        this.tmplList = this.tmplList.filter(t => t !== tmpl);
```

Comment:
I understand what you're trying to do, but tmpl again is an object with a template_id and name property. So I'm not positive this works...